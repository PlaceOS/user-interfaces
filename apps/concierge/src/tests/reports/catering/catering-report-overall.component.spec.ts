import { createComponentFactory, Spectator } from '@ngneat/spectator/jest';
import { BehaviorSubject } from 'rxjs';

import { CateringReportOverallComponent } from 'apps/concierge/src/app/reports/catering/catering-report-overall.component';
import { CateringReportStateService } from 'apps/concierge/src/app/reports/catering/catering-report-state.service';

describe('CateringReportOverallComponent', () => {
    let spectator: Spectator<CateringReportOverallComponent>;
    const createComponent = createComponentFactory({
        component: CateringReportOverallComponent,
        declarations: [],
        providers: [
            {
                provide: CateringReportStateService,
                useValue: {
                    stats: new BehaviorSubject({}),
                },
            },
        ],
    });

    beforeEach(() => (spectator = createComponent()));

    it('should create component', () => {
        expect(spectator.component).toBeTruthy();
    });

    it('should match snapshot', () => {
        expect(spectator.element).toMatchSnapshot();
        (spectator.inject(CateringReportStateService).stats as any).next({
            order_count: 32,
            unique_items: 60,
            item_count: 120,
            total_cost: 1200_00,
            avg_cost: 3000,
        });
        spectator.detectChanges();
        expect(spectator.element).toMatchSnapshot();
    });
});
