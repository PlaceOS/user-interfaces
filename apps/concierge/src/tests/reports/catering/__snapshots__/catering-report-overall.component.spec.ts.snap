// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CateringReportOverallComponent should match snapshot 1`] = `
<div
  id="root1"
>
  <div
    class="bg-white shadow rounded m-2 px-8 w-48 h-24 flex flex-col items-center justify-center"
  >
    <div
      class="text-3xl"
    >
       0 
    </div>
    <div>
      Orders
    </div>
  </div>
  <div
    class="bg-white shadow rounded m-2 px-8 w-48 h-24 flex flex-col items-center justify-center"
  >
    <div
      class="text-3xl"
    >
       0 
    </div>
    <div>
      Unique Items
    </div>
  </div>
  <div
    class="bg-white shadow rounded m-2 px-8 w-48 h-24 flex flex-col items-center justify-center"
  >
    <div
      class="text-3xl"
    >
      0
    </div>
    <div>
      Ordered Items
    </div>
  </div>
  <div
    class="bg-white shadow rounded m-2 px-8 w-48 h-24 flex flex-col items-center justify-center"
  >
    <div
      class="text-3xl"
    >
       $0.00 
    </div>
    <div>
      Total
    </div>
  </div>
  <div
    class="bg-white shadow rounded m-2 px-8 w-48 h-24 flex flex-col items-center justify-center"
  >
    <div
      class="text-3xl"
    >
       $0.00 
    </div>
    <div>
      Order Average
    </div>
  </div>
</div>
`;

exports[`CateringReportOverallComponent should match snapshot 2`] = `
<div
  id="root1"
>
  <div
    class="bg-white shadow rounded m-2 px-8 w-48 h-24 flex flex-col items-center justify-center"
  >
    <div
      class="text-3xl"
    >
       32 
    </div>
    <div>
      Orders
    </div>
  </div>
  <div
    class="bg-white shadow rounded m-2 px-8 w-48 h-24 flex flex-col items-center justify-center"
  >
    <div
      class="text-3xl"
    >
       60 
    </div>
    <div>
      Unique Items
    </div>
  </div>
  <div
    class="bg-white shadow rounded m-2 px-8 w-48 h-24 flex flex-col items-center justify-center"
  >
    <div
      class="text-3xl"
    >
      120
    </div>
    <div>
      Ordered Items
    </div>
  </div>
  <div
    class="bg-white shadow rounded m-2 px-8 w-48 h-24 flex flex-col items-center justify-center"
  >
    <div
      class="text-3xl"
    >
       $1,200.00 
    </div>
    <div>
      Total
    </div>
  </div>
  <div
    class="bg-white shadow rounded m-2 px-8 w-48 h-24 flex flex-col items-center justify-center"
  >
    <div
      class="text-3xl"
    >
       $30.00 
    </div>
    <div>
      Order Average
    </div>
  </div>
</div>
`;
